kbase-extension/static/kbase/js/widgets/function_output/kbaseExpressionGenesetBaseWidget.js

```diff
@@ -258,54 +276,24 @@ define([
             return $row;
         },
         loading: function (isLoading) {
-            if (isLoading) this.showMessage("<img src='" + this.options.loadingImage + "'/>");
-            else this.hideMessage();
+            if (isLoading) {
+                this.showMessage($LoadingMessage('Loading...'));
+            } else {
+                this.hideMessage();
+            }
         },
-        showMessage: function (message) {
-            const span = $('<span/>').append(message);
-
-            this.$messagePane.append(span);
+        showMessage: function ($message) {
+            this.$messagePane.html($message);
             this.$messagePane.show();
         },
         hideMessage: function () {
             this.$messagePane.hide();
             this.$messagePane.empty();
         },
-        clientError: function (error) {
+        renderError: function (error) {
             this.loading(false);
-            let errString = 'Unknown error.';
-            console.error(error);
-            if (typeof error === 'string') errString = error;
-            else if (error.error && error.error.message) errString = error.error.message;
-            else if (error.error && error.error.error && typeof error.error.error === 'string') {
-                errString = error.error.error;
-                if (
-                    errString.indexOf('java.lang.NullPointerException') > -1 &&
-                    errString.indexOf('buildIndeces(KBaseFeatureValuesImpl.java:708)') > -1
-                ) {
-                    // this is a null pointer due to an unknown feature ID.  TODO: handle this gracefully
-                    errString = 'Feature IDs not found.<br><br>';
-                    errString +=
-                        'Currently all Features included in a FeatureSet must be present' +
-                        ' in the Expression Data Matrix.  Please rebuild the FeatureSet ' +
-                        'so that it only includes these features.  This is a known issue ' +
-                        'and will be fixed shortly.';
-                }
-            }
-
-            const $errorDiv = $('<div>')
-                .addClass('alert alert-danger')
-                .append('<b>Error:</b>')
-                .append('<br>' + errString);
             this.$elem.empty();
-            this.$elem.append($errorDiv);
-        },
-        uuid: function () {
```

Comment:
It makes me happy to see individual module reinventions of uuids go away. 😄

Reply:
yeah, I try to zap the uuid method whenever I need to touch file in a significant way. There has been a proper ui module in the codebase for quite a while now (but not originally, of course!).